onadata/apps/logger/tests/data_migration/test_survey_tree.py

```diff
@@ -136,5 +136,6 @@ def test_insert_field_into_group_chain__real_world_scenario(self):
         for field, groups in zip(fields, fields_groups):
             self.survey_prev.insert_field_into_group_chain(field, groups)
 
-        self.assertXMLsEqual(self.survey_prev.to_string(),
-                             fixtures.survey_xml_groups_after__second)
+        expected_xml = self.survey_prev.to_string()\
+            .replace('TestGroup', 'AlgebraicTypes2')
```

Comment:
Why `.replace` is used there? My guess is that `fixtures.survey_xml_groups_before__second` should have had replaced `'TestGroup'` with `'AlgebraisTypes2'`, otherwise it seems like migration tool is not doing some important step.

Reply:
It's rather the opposite + it's not migration step yet, it's just a class for handling tree operations. I have `old` xml and `new` one - in this test i'm operating on `old` one in such a way, that it should result in new one, although, since i'm not really migrating it, the root node stays the same, thus, i need to replace it so that it match newer one.